traits/src/lib.rs

```diff
@@ -86,3 +86,30 @@ impl<T> Handler<T> for Tuple {
 		Ok(())
 	}
 }
+
+pub trait PreConditions<T> {
```

Comment:
this is basically the `Filter` trait from Substrate

Reply:
Yeah, but parity seems to abandon this one: https://github.com/paritytech/substrate/blob/3168e10f6576a28930554bbf903ff9900b833a14/frame/support/src/traits/members.rs#L44-L51